packages/nimble-components/src/rich-text-editor-viewer/specs/README.md

```diff
@@ -0,0 +1,335 @@
+# Nimble Rich Text Editor and Nimble Rich Text Viewer
+
+## Overview
+
+-   The `nimble-rich-text-editor` component is a text editing tool that offers a comprehensive range of functionalities for
+    incorporating rich text content. It provides users with the ability to apply different text styles and formats to their content.
+-   The `nimble-rich-text-viewer` component is way to view all the rich text content.
+
+### Background
+
+[Nimble issue #1288](https://github.com/ni/nimble/issues/1288)
+
+[Comments UI mockup](https://www.figma.com/file/Q5SU1OwrnD08keon3zObRX/SystemLink?type=design&node-id=6280-94045)
+
+### Non-goals
+
+-   The visual and interaction design for `nimble-rich-text-editor` and `nimble-rich-text-viewer` is out of scope for this spec component.
+-   Blazor integration will be out of scope for this component as we have less or no experience in Blazor component development or other
+    related technology. Blazor example app addition will also be not covered in this spec.
+-   Adding an example Angular client application for this component may not be included in the initial releases. However, the integration
+    of the rich-text-editor with Angular applications, achieved through a directive, will be developed following the completion of this
+    component.
+
+### Features
+
+The `nimble-rich-text-editor` supports the basic text formatting options that includes:
+
+1. Bold
+2. Italics
+3. Strikeout
+4. Numbered List
+5. Bulleted List
+
+This component will also offer APIs and interactive methods to format texts in the following ways:
+
+1. Using the buttons located below the editor to modify the text formatting for the selected texts.
+2. Using the keyboard shortcuts to modify the text formatting in the editor.
+3. Support to get markdown output from the editor.
+4. Support to convert the markdown to HTML to render it back in the `nimble-rich-text-viewer` component.
+
+#### _Additional features out of scope of this spec_
+
+-   Allowing the user to tag or mention by entering `@` in the editor and selecting the user name from the drop-down list.
+-   Support for adding images to the editor either by uploading or by pasting it.
+-   Support for adding hyperlinks to the existing words in the editor. However, adding raw links by pasting to the editor is supported.
+-   Support for underlining the text.
```

Comment:
Since we are putting underline support out of scope which is typically done using `<u>` tag of HTML in any markdown, this eliminates our dependency to use html within markdown for current set of features. 

If there are options in tip tap to switch off HTML tags in markdown, it will be worthy to use it and mention it in security section which will completely disallow injection attacks using `<script>` tags into our component, until we need HTML tags

Reply:
Deferring this comment as it is captured in the security section of this doc.